packages/server/graphql/intranetSchema/mutations/hardDeleteUser.ts

```diff
@@ -0,0 +1,313 @@
+      r
+        .table('AgendaItem')
+        .getAll(r.args(teamIds), {index: 'teamId'})
+        .filter((row) => r(teamMemberIds).contains(row('teamMemberId')))
+        .filter((row) => row('isActive').eq(false))
+        .getField('id')
+        .coerceTo('array')
+        .distinct()
+        .run(),
+      r
+        .table('AgendaItem')
+        .getAll(r.args(teamIds), {index: 'teamId'})
+        .filter((row) => r(teamMemberIds).contains(row('teamMemberId')))
+        .filter((row) => row('isActive').eq(true))
+        .getField('id')
+        .coerceTo('array')
+        .distinct()
+        .run(),
+      (r
+        .table('NewMeeting')
+        .getAll(r.args(teamIds), {index: 'teamId'})
+        .eqJoin('id', r.table('RetroReflection'), {index: 'meetingId'})
+        .zip() as any)
+        .filter((row) => row('creatorId').eq(r(userIdToDelete)))
+        .getField('id')
+        .coerceTo('array')
+        .distinct()
+        .run(),
+      r
+        .table('NewMeeting')
+        .getAll(r.args(teamIds), {index: 'teamId'})
+        .filter((row) => row('facilitatorUserId').eq(r(userIdToDelete)))
+        .merge((meeting) => ({
+          otherTeamMember: r
+            .table('TeamMember')
+            .getAll(meeting('teamId'), {index: 'teamId'})
+            .filter((row) => row('userId').ne(r(userIdToDelete)))
+            .nth(0)
+            .getField('userId')
+            .default(null)
+        }))
+        .filter(r.row.hasFields('otherTeamMember'))
+        .pluck('id', 'otherTeamMember')
+        .run(),
+      r
+        .table('NewMeeting')
+        .getAll(r.args(teamIds), {index: 'teamId'})
+        .filter((row) => row('createdBy').eq(r(userIdToDelete)))
+        .merge((meeting) => ({
+          otherTeamMember: r
+            .table('TeamMember')
+            .getAll(meeting('teamId'), {index: 'teamId'})
+            .filter((row) => row('userId').ne(r(userIdToDelete)))
+            .nth(0)
+            .getField('userId')
+            .default(null)
+        }))
+        .filter(r.row.hasFields('otherTeamMember'))
+        .pluck('id', 'otherTeamMember')
+        .run(),
+      pg.query(`SELECT "id" FROM "Discussion" WHERE "teamId" = ANY ($1);`, [teamIds])
+    ])
+    const discussionIds = discussions.rows.map(({id}) => id)
+
+    // soft delete first for side effects, then completely remove user dependent stages
+    await softDeleteUser(user, dataLoader, reason, true)
+
+    for (const teamMemberId of teamMemberIds) {
+      const teamId = teamMemberId.split('::')[1]
+      const filterFn = (stage: CheckInStage | UpdatesStage | EstimateStage) =>
+        (stage as CheckInStage | UpdatesStage).teamMemberId === teamMemberId ||
+        (stage as EstimateStage).creatorUserId === userIdToDelete
+      await removeStagesFromMeetings(filterFn, teamId, dataLoader, true)
+      await removeUserFromMeetingStages(userIdToDelete, teamId, dataLoader, true)
+    }
+    for (const agendaItemId of currentAgendaItemIds) {
```

Comment:
cool that makes sense. i usually try to err on the side of not changing existing functions, since they might be written that way for a specific reason i don't know about. moving the logic into `softDeleteUser` definitely simplifies things!